fetch.bs

```diff
@@ -3614,83 +3668,96 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
+    via HTTP/1.0 or older, then return a <a>network error</a>.
+
+    <p>If the HTTP request results in a TLS client certificate dialog, run these substeps:
+
+    <ol>
+     <li><p>If <var>request</var>'s <a for=request>window</a>
+     is an <a>environment settings object</a>, make the dialog
+     available in <var>request</var>'s
+     <a for=request>window</a>.
 
-   <li><p>Otherwise, return a <a>network error</a>.
+     <li><p>Otherwise, return a <a>network error</a>.
+    </ol>
+
+    <p>If <var>response</var> was retrieved over HTTPS, set its
+    <a for=response>HTTPS state</a> to either
+    "<code>deprecated</code>" or "<code>modern</code>".
+    [[!TLS]]
+
+    <p class="note no-backref">The exact determination here is up to user agents for the
+    time being. User agents are strongly encouraged to only succeed HTTPS connections with
+    strong security properties and return
+    <a>network errors</a> otherwise. Using the
+    "<code>deprecated</code>" state value ought to be a temporary and last resort kind
+    of option.
+
+    <p><a for=request>Transmit body</a> for <var>request</var>.
   </ol>
 
-  <p>If <var>response</var> was retrieved over HTTPS, set its
-  <a for=response>HTTPS state</a> to either
-  "<code>deprecated</code>" or "<code>modern</code>".
-  [[!TLS]]
+ <li>
+  <p>If fetch is <a for=fetch>terminated</a>, providing a <var>reason</var>, then:
+
+  <ol>
+   <li><p>If <var>connection</var> is an HTTP/2 connection, then transmit an
+   "<code>RST_STREAM</code>" to cancel the underlying stream.
 
-  <p class="note no-backref">The exact determination here is up to user agents for the
-  time being. User agents are strongly encouraged to only succeed HTTPS connections with
-  strong security properties and return
-  <a>network errors</a> otherwise. Using the
-  "<code>deprecated</code>" state value ought to be a temporary and last resort kind
-  of option.
+   <li><p>Otherwise, close <var>connection</var>.
 
-  <p><a for=request>Transmit body</a> for <var>request</var>.
+   <li><p>Return a <a>network error</a> with <a for=response>termination reason</a> set to
+   <var>reason</var>.
+  </ol>
 
  <li>
   <p>Let <var>strategy</var> be an object. The user agent may choose any object.
 
   <p class="note no-backref"><var>strategy</var> is used to control the queuing strategy of
   <var>stream</var> constructed below.
 
- <li><p>Let <var>pull</var> be an action that <a lt=resumed for=fetch>resumes</a> the
- ongoing fetch if it is <a lt=suspend for=fetch>suspended</a>.
+ <li><p>Let <var>pull</var> be an action that <a lt=resumed for=fetch>resumes</a> the ongoing fetch
+ if it is <a lt=suspend for=fetch>suspended</a>.
 
- <li><p>Let <var>cancel</var> be an action that
- <a lt=terminated for=fetch>terminates</a> the ongoing fetch with reason
- <i>end-user abort</i>.
+ <li><p>Let <var>cancel</var> be an action that <a lt=terminated for=fetch>terminates</a> the
+ ongoing fetch with reason <i>fatal</i>.
```

Comment:
Closing a tab is not what's in play here. This is about the user calling .cancel() on a stream they are reading. Or, because it goes through the same JS API, the browser calling https://fetch.spec.whatwg.org/branch-snapshots/cancelation/#concept-cancel-readablestream .

That is called by https://fetch.spec.whatwg.org/branch-snapshots/cancelation/#abort-fetch , in fact. So this is saying that any fetch which is aborted through an AbortController will [manifest to XHR](https://xhr.spec.whatwg.org/#handle-errors) not as an abort, but instead as a network error with (I believe) no events fired.

I don't think this is right, in short.